src/NServiceBus.Autofac/AutofacObjectBuilder.cs

```diff
@@ -48,88 +69,85 @@ void DisposeManaged()
                 return;
             }
             container?.Dispose();
+
+            (builder as ParentBuilder)?.parentScope?.Dispose();
         }
 
         public Common.IContainer BuildChildContainer()
         {
-            return new AutofacObjectBuilder(container.BeginLifetimeScope(), true, true);
+            var childScope = Container.BeginLifetimeScope();
+
+            return new AutofacObjectBuilder(childScope.CreateBuilderFromContainer(true), true, true);
```

Comment:
We could do it all in one line.

The `childScope` was just a var so I didn't have to chain too many method calls together.

`public Common.IContainer BuildChildContainer() =>
    new AutofacObjectBuilder(Container.BeginLifetimeScope().CreateBuilderFromContainer(true), true, true);`